coderd/database/querier_test.go

```diff
@@ -494,6 +494,164 @@ func TestUserChangeLoginType(t *testing.T) {
 	require.Equal(t, bobExpPass, bob.HashedPassword, "hashed password should not change")
 }
 
+type tvArgs struct {
+	Status database.ProvisionerJobStatus
+	// CreateWorkspace is true if we should create a workspace for the template version
+	CreateWorkspace     bool
+	WorkspaceTransition database.WorkspaceTransition
+}
+
+func createTemplateVersion(t testing.TB, db database.Store, tpl database.Template, args tvArgs) database.TemplateVersion {
+	version := dbgen.TemplateVersion(t, db, database.TemplateVersion{
+		TemplateID: uuid.NullUUID{
+			UUID:  tpl.ID,
+			Valid: true,
+		},
+		OrganizationID: tpl.OrganizationID,
+		CreatedAt:      dbtime.Now(),
+		UpdatedAt:      dbtime.Now(),
+		CreatedBy:      tpl.CreatedBy,
+	})
+
+	earlier := sql.NullTime{
+		Time:  dbtime.Now().Add(time.Second * -30),
+		Valid: true,
+	}
+	now := sql.NullTime{
+		Time:  dbtime.Now(),
+		Valid: true,
+	}
+	j := database.ProvisionerJob{
+		ID:             version.JobID,
+		CreatedAt:      earlier.Time,
+		UpdatedAt:      earlier.Time,
+		Error:          sql.NullString{},
+		OrganizationID: tpl.OrganizationID,
+		InitiatorID:    tpl.CreatedBy,
+		Type:           database.ProvisionerJobTypeTemplateVersionImport,
+	}
+
+	switch args.Status {
+	case database.ProvisionerJobStatusRunning:
+		j.StartedAt = earlier
+	case database.ProvisionerJobStatusPending:
+	case database.ProvisionerJobStatusFailed:
+		j.StartedAt = earlier
+		j.CompletedAt = now
+		j.Error = sql.NullString{
+			String: "failed",
+			Valid:  true,
+		}
+		j.ErrorCode = sql.NullString{
+			String: "failed",
+			Valid:  true,
+		}
+	case database.ProvisionerJobStatusSucceeded:
+		j.StartedAt = earlier
+		j.CompletedAt = now
+	default:
+		t.Fatalf("invalid status: %s", args.Status)
+	}
+
+	dbgen.ProvisionerJob(t, db, nil, j)
+	if args.CreateWorkspace {
+		wrk := dbgen.Workspace(t, db, database.Workspace{
+			CreatedAt:      time.Time{},
+			UpdatedAt:      time.Time{},
+			OwnerID:        tpl.CreatedBy,
+			OrganizationID: tpl.OrganizationID,
+			TemplateID:     tpl.ID,
+		})
+		trans := database.WorkspaceTransitionStart
+		if args.WorkspaceTransition != "" {
+			trans = args.WorkspaceTransition
+		}
+		buildJob := dbgen.ProvisionerJob(t, db, nil, database.ProvisionerJob{
+			Type:           database.ProvisionerJobTypeWorkspaceBuild,
+			CompletedAt:    now,
+			InitiatorID:    tpl.CreatedBy,
+			OrganizationID: tpl.OrganizationID,
+		})
+		dbgen.WorkspaceBuild(t, db, database.WorkspaceBuild{
+			WorkspaceID:       wrk.ID,
+			TemplateVersionID: version.ID,
+			BuildNumber:       1,
+			Transition:        trans,
+			InitiatorID:       tpl.CreatedBy,
+			JobID:             buildJob.ID,
+		})
+	}
+	return version
+}
+
+func TestArchiveVersions(t *testing.T) {
```

Comment:
The cli/api branch has a test at the coderd layer, adding this test at the db layer since the coderd layer of this feat does not exist yet.

Reply:
I think it's no harm to have some database-layer tests.